addons/website_slides/models/slide_channel.py

```diff
@@ -956,18 +956,20 @@ def _action_add_members(self, target_partners, member_status='joined', raise_on_
                 )
         return result_channel_partners
 
-    def _filter_add_members(self, target_partners, raise_on_access=False):
+    def _filter_add_members(self, raise_on_access=False):
         allowed = self.filtered(lambda channel: channel.enroll == 'public')
-        on_invite = self.filtered(lambda channel: channel.enroll == 'invite')
-        if on_invite:
-            try:
-                on_invite.check_access_rights('write')
-                on_invite.check_access_rule('write')
-            except AccessError:
-                if raise_on_access:
-                    raise AccessError(_('You are not allowed to add members to this course. Please contact the course responsible or an administrator.'))
-            else:
-                allowed |= on_invite
+        if controlled_access := self - allowed:
+            # If not raise, check each type independently as they may have different rules.
+            batches = [controlled_access] if raise_on_access else controlled_access.grouped('enroll').values()
+            for controlled_access_with_enroll in batches:
```

Comment:
`check_access_rights` can be called once because it just depends on the model, will something like that work ?
```py
batches.check_access_rights('write')
allowed = batches._filter_access_rules('write')
if batches - allowed and raise_on_access:
    raise AccessError(_('You are not allowed to add members to this course. Please contact the course responsible or an administrator.'))
```
(assuming batches is a recordset and not a list)